File: src/types/subscription.ts
export interface SubscriptionStatus {
  is_pro: boolean;
  plan_name?: string | null;
  status?: string | null;
  amount?: number | null;
  start_date?: string | null;
  expiry_date?: string | null;
  renewal_date?: string | null;
  subscription_id?: string | null;
}

export interface CreateOrderResponse {
  order_id: string;
  amount: number;
  currency: string;
  key_id: string;
  plan_name: string;
  subscription_id: string;
}

export interface PaymentHistoryItem {
  id: string;
  amount: number;
  payment_method: string | null;
  transaction_id: string | null;
  status: string;
  created_at: string;
  plan_name?: string | null;
}

export interface SubscriptionHistoryItem {
  id: string;
  plan_name: string;
  amount: number;
  status: string;
  start_date: string | null;
  expiry_date: string | null;
  created_at: string;
}

export interface SubscriptionHistory {
  subscriptions: SubscriptionHistoryItem[];
  payments: PaymentHistoryItem[];
}

export interface VerifyPaymentPayload {
  razorpay_order_id: string;
  razorpay_payment_id: string;
  razorpay_signature: string;
}

File: backend/app/schemas/subscription.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class CreateOrderResponse(BaseModel):
    order_id: str
    amount: int
    currency: str = "INR"
    key_id: str
    plan_name: str
    subscription_id: UUID


class VerifyPaymentRequest(BaseModel):
    razorpay_order_id: str = Field(min_length=3)
    razorpay_payment_id: str = Field(min_length=3)
    razorpay_signature: str = Field(min_length=3)


class SubscriptionStatusResponse(BaseModel):
    is_pro: bool
    plan_name: str | None = None
    status: str | None = None
    amount: int | None = None
    start_date: datetime | None = None
    expiry_date: datetime | None = None
    renewal_date: datetime | None = None
    subscription_id: UUID | None = None


class PaymentHistoryItem(BaseModel):
    id: UUID
    amount: int
    payment_method: str | None
    transaction_id: str | None
    status: str
    created_at: datetime
    plan_name: str | None = None

    model_config = {"from_attributes": True}


class SubscriptionHistoryItem(BaseModel):
    id: UUID
    plan_name: str
    amount: int
    status: str
    start_date: datetime | None
    expiry_date: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class SubscriptionHistoryResponse(BaseModel):
    subscriptions: list[SubscriptionHistoryItem]
    payments: list[PaymentHistoryItem]
